File: go-client-sdk/pkg/models/operations/getcompaniescompanyidreportsevents.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
	"time"
)

type GetCompaniesCompanyIDReportsEventsPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
}

type GetCompaniesCompanyIDReportsEventsQueryParams struct {
	FromDate *time.Time `queryParam:"style=form,explode=true,name=fromDate"`
	PageSize *int       `queryParam:"style=form,explode=true,name=pageSize"`
	ToDate   *time.Time `queryParam:"style=form,explode=true,name=toDate"`
}

type GetCompaniesCompanyIDReportsEventsSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDReportsEventsRequest struct {
	PathParams  GetCompaniesCompanyIDReportsEventsPathParams
	QueryParams GetCompaniesCompanyIDReportsEventsQueryParams
	Security    GetCompaniesCompanyIDReportsEventsSecurity
}

type GetCompaniesCompanyIDReportsEventsResponse struct {
	CodatPublicAPIModelsCompanyCompanyEventStream *shared.CodatPublicAPIModelsCompanyCompanyEventStream
	ContentType                                   string
	StatusCode                                    int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyidreportsevents.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Transform, Type } from "class-transformer";


export class GetCompaniesCompanyIdReportsEventsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;
}

export class GetCompaniesCompanyIdReportsEventsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fromDate;dateTimeFormat=YYYY-MM-DDThh:mm:ss.sssZ" })
  fromDate?: Date;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=pageSize" })
  pageSize?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=toDate;dateTimeFormat=YYYY-MM-DDThh:mm:ss.sssZ" })
  toDate?: Date;
}

export class GetCompaniesCompanyIdReportsEventsSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetCompaniesCompanyIdReportsEventsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdReportsEventsPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCompaniesCompanyIdReportsEventsQueryParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdReportsEventsSecurity;
}

export class GetCompaniesCompanyIdReportsEventsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatPublicApiModelsCompanyCompanyEventStream?: shared.CodatPublicApiModelsCompanyCompanyEventStream;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}